internal/target/dtrack_target.go

```diff
@@ -0,0 +1,75 @@
+package target
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+
+	parser "github.com/novln/docker-parser"
+	dtrack "github.com/nscuro/dtrack-client"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+
+	"github.com/ckotzbauer/sbom-operator/internal"
+	"github.com/ckotzbauer/sbom-operator/internal/kubernetes"
+)
+
+type DependencyTrackTarget struct {
+	baseUrl string
+	apiKey  string
+}
+
+func NewDependencyTrackTarget() *DependencyTrackTarget {
+	baseUrl := viper.GetString(internal.ConfigKeyDependencyTrackBaseUrl)
+	apiKey := viper.GetString(internal.ConfigKeyDependencyTrackApiKey)
+	return &DependencyTrackTarget{
+		baseUrl: baseUrl,
+		apiKey:  apiKey,
+	}
+}
+
+func (g *DependencyTrackTarget) ValidateConfig() error {
+	if g.baseUrl == "" {
+		return fmt.Errorf("%s is empty", internal.ConfigKeyDependencyTrackBaseUrl)
+	}
+	if g.apiKey == "" {
+		return fmt.Errorf("%s is empty", internal.ConfigKeyDependencyTrackApiKey)
+	}
+	return nil
+}
+
+func (g *DependencyTrackTarget) Initialize() {
+}
+
+func (g *DependencyTrackTarget) ProcessSbom(image kubernetes.ContainerImage, sbom string) {
```

Comment:
I am not yet too familiar with this project but it feels like this only makes sense for the processing of SBOMs for container images.   If you had a second use case for the operator that pulled binaries from an artifact repo, for example, the ProcessSbom contract wouldn't fit.

It might pay to move the parsing of the docker image name up to the caller and require the name and version on this interface.  I suggest the change in terminology too to more fit with the intended target  - imageName and imageTag are used later as projectName and projectVersion when integrating with DT so why not be specific about that?

Of course this works fine as it it - so i just a suggestion.

Reply:
Not sure how that would affect the GitTarget though as it is using a different part of the image metadata (the ID) to create a path.  So may be something to consider as a future improvement.

Reply:
Thanks for that feedback. Currently there are no plans to use the operator for other artifacts, so for now the containerImage fits the requirements.